File: C++ WorldSimulatorGame/POproj1/WilczeJagody.cpp
#include "WilczeJagody.h"

WilczeJagody::WilczeJagody(Punkt polozenie, Swiat* swiat)
	:Roslina(99, 0, 'J', 0, true, Organizmy::wilczeJagody, polozenie, swiat) {
	this->sila = 99;
	this->inicjatywa = 0;
	this->znak = 'J';
	this->wiek = 0;
}
string WilczeJagody::getNazwaOrganizmu() {
	return "Wilcze Jagody";
}
void WilczeJagody::kolizja(Organizm* other, bool atakujacy, bool* ruch) {

	if (this->getSila() > other->getSila() || (this->getSila() == other->getSila() && atakujacy)) {
		int tmp = this->sila;
		this->setSila(other->getSila() + 10);
		other->kolizja(this, false, ruch);
		this->setSila(tmp);
	}
	else {
		zabojczeInformacjeDlaRoslin(other);
		//std::cout << other->getZnak() << " z sila: " << other->getSila() << " zjadl -> " << this->getZnak() << " z sila: " << this->getSila() << std::endl;
		swiat->wyczyscOrganizmZMapy(this->polozenie);
		//swiat->ZabijOrganizm(other);
		//other->kolizja(this, false);
	}
}

File: Python WorldSimulatorGame/WilczeJagody.py
from Organizm import Organizm
from Roslina import Roslina


class WilczeJagody(Roslina):

    def __init__(self, polozenie, swiat):
        super(WilczeJagody, self).__init__(99, 0, 'J', 0, True, polozenie, swiat)

    def getNazwaOrganizmu(self):
        return "Wilcze Jagody"

    def kolizja(self, other: Organizm, atakujacy, ruch):
            self.zabojczeInformacjeDlaRoslin(other, self)
            self.setSila(other.getSila()+10)
            self.swiat.setPrzerwij(self.swiat.getPrzerwij() + 1)
            other.kolizja(self, True, ruch)
            self.setZycie(False)


